server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatEventParser.ts

```diff
@@ -130,10 +130,19 @@ export class AgenticChatEventParser implements ChatResult {
                 }
 
                 if (toolUseEvent.stop) {
-                    const parsedInput =
-                        typeof this.toolUses[toolUseId].input === 'string'
-                            ? JSON.parse(this.toolUses[toolUseId].input === '' ? '{}' : this.toolUses[toolUseId].input)
-                            : this.toolUses[toolUseId].input
+                    let parsedInput
+                    try {
+                        parsedInput =
```

Comment:
theres a lot of conditional logic in this, can we unwind it so it makes it a little easier to understand